pages/index.tsx

```diff
@@ -16,5 +12,17 @@ const IndexPage: React.FC<{}> = () => {
     </>
   )
 }
+export const getServerSideProps: GetServerSideProps = async context => {
+  return context.req.cookies['connect.sid']
+    ? {
+        redirect: {
+          destination: '/curriculum',
+          permanent: false
+        }
+      }
```

Comment:
How about having another path for the landing page, like `/home` or `/landing`, and keep this `index` page only for redirection. This way we can keep the static page benefits while having the server-side redirection.

Reply:
I could do it but I don't see the benefits. I'm not sure how exactly next handles server side redirects though. Does it automatically sends back redirected page or it sends back some redirect object first? 

Currently it's request to index->check for cookie->returning landing page. With redirect it would be request to index->check for cookie->returning redirect object->go to static landing page->returning landing page. Even if next automatically sends back curriculum page it would be about the same speed wise. Only people who would go straight to landing page would benefit from static optimization.

Reply:
I'm just concerned because we'll lose the static page generation benefits we have now, since using `getServerSideProps` forces the page to be server-side rendered. And making the landing page SSR seems wasteful because it is 100% static content (except for the profile query on the nav bar of course).

Unfortunately I don't know a good solution for this yet, but in my opinion we gotta think this through for some more time.